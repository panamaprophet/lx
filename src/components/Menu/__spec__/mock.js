module.exports = [{
	id: 4,
	name: 'space',
}, {
	id: 2,
	name: 'galaxy',
}, {
	id: 0,
	name: 'universe'
}];