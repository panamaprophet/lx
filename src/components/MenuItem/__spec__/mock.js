const linear = {
	id: 420,
	name: 'clouds',
	isActive: false,
	onClick: () => {},
	
};

const withSubtree = {
	...linear,
	isActive: true,
	items: [{
		id: 4,
		name: 'space',
	}, {
		id: 2,
		name: 'galaxy',
	}, {
		id: 0,
		name: 'universe'
	}],
	SubMenuComponent: () => 'Component of Submenu',
}

module.exports = {
	linear,
	withSubtree,
};