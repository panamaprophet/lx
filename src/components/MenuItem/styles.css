.root {
    background: #009688;
}

.isActive > .title {
    background: #ffeb3b;
}

.title {
    padding: 8px;
    cursor: pointer;
}

.collapsed {
    display: none;
}

.expanded {
    display: initial;
}