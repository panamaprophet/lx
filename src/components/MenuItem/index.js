import React from 'react';
import cx from 'classnames';

import styles from './styles.css';

const MenuItem = ({
    id,
    name,
    items,
    isActive = false,
    onClick = () => {},
    SubMenuComponent,
}) => (
    <li className={cx(styles.root, {
        [styles.isActive]: isActive,
    })}>
        <div className={styles.title} onClick={() => onClick(id)}>
            {name}
        </div>
        {isActive && items && SubMenuComponent && <SubMenuComponent items={items} />}
    </li>
);

export default MenuItem;
