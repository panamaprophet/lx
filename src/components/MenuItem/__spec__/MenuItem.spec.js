import React from 'react';
import {shallow} from 'enzyme';

import MenuItem from '../index';
import {linear, withSubtree} from './mock';

describe('Menu Item', () => {
    test('component renders with one-level menu', () => {
        const component = shallow(<MenuItem 
            {...linear} 
        />);

        expect(component).toMatchSnapshot();
    });

    test('component renders with active submenu', () => {
        const component = shallow(<MenuItem 
            {...withSubtree} 
        />);

        expect(component).toMatchSnapshot();
    });

    test('component triggers onClick by click', () => {
        const onClick = jest.fn();

        const component = shallow(<MenuItem 
            onClick={onClick}
        />)

        component.find('div').simulate('click');

        expect(onClick).toBeCalled();
    });
});