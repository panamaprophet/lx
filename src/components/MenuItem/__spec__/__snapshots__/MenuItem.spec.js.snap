// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item component renders with active submenu 1`] = `
<li
  className="root isActive"
>
  <div
    className="title"
    onClick={[Function]}
  >
    clouds
  </div>
  <SubMenuComponent
    items={
      Array [
        Object {
          "id": 4,
          "name": "space",
        },
        Object {
          "id": 2,
          "name": "galaxy",
        },
        Object {
          "id": 0,
          "name": "universe",
        },
      ]
    }
  />
</li>
`;

exports[`Menu Item component renders with one-level menu 1`] = `
<li
  className="root"
>
  <div
    className="title"
    onClick={[Function]}
  >
    clouds
  </div>
</li>
`;
